content/dart/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,255 @@
+
+```
+
+String identifierName = 'myIdentifier';
+Symbol badSymbol = Symbol(identifierName);
+
+print(badSymbol);  // Outcome: Error, minification can impact identifier symbols
+
+```
+
+## Lists
+
+List ``List`` is used to represent a collection of ordered group of objects, working simillar to `arrays` in other programming languages.
+
+```
+
+List<int> numbersList = [1, 2, 3, 4, 5];
+
+print(numbersList);  // Output: [1, 2, 3, 4, 5]
+
+```
+
+Below is an example of incorrect declarations of values to ``List`` data type variables.
+
+```
+
+List<String> invalidList = [1, 2, 3];  // Incorrect: Mismatched types in the list
+
+print(invalidList);  // Outcome: Error, incompatible types in the list
+
+```
+
+## Maps
+
+Map ``Map`` object consists in a key and value pair and they can be any data type.
+
+```
+
+Map<String, int> studentScores = {
+  'Alice': 95,
+  'Bob': 87,
+  'Charlie': 92,
+};
+
+print(studentScores['Alice']);  // Output: 95
+print(studentScores['Bob']);    // Output: 87
+
+```
+
+Below is of example with incorrect or not so recommended declarations of values to `Map` data type variable.
+
+```
+Map<int, String> invalidMap = {
+    'John': 'Doe'};  // Mismatched key-value types
+
+Map<dynamic, dynamic> dynamicMap = {
+    'name': 'Alice', 42: 'age'};  // Works, but not recommended for type safety 
+
+print(invalidMap['John']);  // Outcome: Error, incompatible key type
+print(dynamicMap[42]);  // Output: age (not recommended)
+
+```
+
+## Nulls
+
+Nulls `Null`is a type that represents the absence of a value in a variable. This type can only have one possible value, which is null. Dart uses a sound null safety system, meaning that variables are non-nullable by default, and they have to be explicitly declared as nullable if they need to be able to hold a null value.
+
+```
+
+int? nullableInt;  // Nullable integer
+String? nullableString;  // Nullable string
+
+print(nullableInt);  // Output: null
+print(nullableString);  // Output: null
+
+```
+
+## Runes
+
+Runes ``Runes`` expose the Unicode code points of a string. Unicode is a character encoding standard that aims to encompass all characters used by humans for writing.
+
+```
+
+String greeting = 'Hello 👋'; // String with emoji
+
+// Using Runes to iterate over the Unicode code points
+for (var codePoint in greeting.runes) {
+  print('Code Point: $codePoint, Character: ${String.fromCharCode(codePoint)}');
+}
+
+```
+
+Below is an example of incorrect declarations of values to `Runes` data type variable.
+
+```
+
+String invalidString = 'Invalid 😞'; // String with emoji
+
+// Attempting to iterate over Unicode code points without using Runes
+for (var character in invalidString.runes) {
+  print('Character: $character');
+}
+
+```
+
+## What is Dynamic?
+
+Dynamic typing is checked dynamically at runtime rather than statically at compile-time. It introduces flexibility but at the same time sacrifices some of the benefits of static typing
```

Comment:
```suggestion
Dynamic typing is checked dynamically at runtime rather than statically at compile-time. It introduces flexibility but at the same time sacrifices some of the benefits of static typing:
```